plugin/pkg/admission/webhook/admission.go

```diff
@@ -0,0 +1,164 @@
+	"k8s.io/apiserver/pkg/util/webhook"
+
+	"k8s.io/kubernetes/pkg/api"
+	admissionv1alpha1 "k8s.io/kubernetes/pkg/apis/admission/v1alpha1"
+
+	// install the clientgo admissiony API for use with api registry
+	_ "k8s.io/kubernetes/pkg/apis/admission/install"
+)
+
+var (
+	groupVersions = []schema.GroupVersion{
+		admissionv1alpha1.SchemeGroupVersion,
+	}
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register("GenericAdmissionWebhook", func(configFile io.Reader) (admission.Interface, error) {
+		var gwhConfig struct {
+			WebhookConfig GenericAdmissionWebhookConfig `json:"webhook"`
+		}
+
+		d := yaml.NewYAMLOrJSONDecoder(configFile, 4096)
+		err := d.Decode(&gwhConfig)
+
+		if err != nil {
+			return nil, err
+		}
+
+		plugin, err := NewGenericAdmissionWebhook(&gwhConfig.WebhookConfig)
+
+		if err != nil {
+			return nil, err
+		}
+
+		return plugin, nil
+	})
+}
+
+// NewGenericAdmissionWebhook returns a generic admission webhook plugin.
+func NewGenericAdmissionWebhook(config *GenericAdmissionWebhookConfig) (admission.Interface, error) {
+	err := normalizeConfig(config)
+
+	if err != nil {
+		return nil, err
+	}
+
+	gw, err := webhook.NewGenericWebhook(api.Registry, api.Codecs, config.KubeConfigFile, groupVersions, config.RetryBackoff)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &GenericAdmissionWebhook{
+		Handler: admission.NewHandler(admission.Connect, admission.Create, admission.Delete, admission.Update),
+		webhook: gw,
+		rules:   config.Rules,
+	}, nil
+}
+
+// GenericAdmissionWebhook is an implementation of admission.Interface.
+type GenericAdmissionWebhook struct {
+	*admission.Handler
+	webhook *webhook.GenericWebhook
+	rules   []Rule
+}
+
+// Admit makes an admission decision based on the request attributes.
+func (a *GenericAdmissionWebhook) Admit(attr admission.Attributes) (err error) {
+	var matched *Rule
+
+	// Process all declared rules to attempt to find a match
+	for i, rule := range a.rules {
+		if Matches(rule, attr) {
+			glog.V(2).Infof("rule at index %d matched request", i)
+			matched = &a.rules[i]
+			break
+		}
+	}
+
+	if matched == nil {
+		glog.V(2).Infof("rule explicitly allowed the request: no rule matched the admission request")
+		return nil
+	}
+
+	// The matched rule skips processing this request
+	if matched.Type == Skip {
+		glog.V(2).Infof("rule explicitly allowed the request")
+		return nil
+	}
+
+	// Make the webhook request
+	request := admissionv1alpha1.NewAdmissionReview(attr)
+	response := a.webhook.RestClient.Post().Body(&request).Do()
+
+	// Handle webhook response
+	if err := response.Error(); err != nil {
+		return a.handleError(attr, matched.FailAction, err)
+	}
+
+	var statusCode int
+	if response.StatusCode(&statusCode); statusCode < 200 || statusCode >= 300 {
+		return a.handleError(attr, matched.FailAction, fmt.Errorf("error contacting webhook: %d", statusCode))
+	}
+
+	if err := response.Into(&request); err != nil {
```

Comment:
This should all be in the one-liner - no need to get so verbose here.  Use the fluent style with rest client.